addons/web/static/src/views/view_components/view_scale_selector.xml

```diff
@@ -18,7 +18,7 @@
             <div t-if="this.props.isWeekendVisible !== undefined" class="dropdown-divider" role="separator"/>
             <DropdownItem
                     t-if="this.props.isWeekendVisible !== undefined"
-                    class="`${props.isWeekendVisible ? 'active' : ''}`"
+                    class="`show_weekends ${props.isWeekendVisible ? 'active' : ''} ${props.currentScale === 'day' ? 'disabled' : ''}`"
```

Comment:
sorry I didn't spot this earlier, but by convention odoo classnames are prefixed by `o_`